lib/webfuse/adapter/impl/filesystem.c

```diff
@@ -16,55 +26,183 @@ static struct fuse_lowlevel_ops const filesystem_operations =
 	.read	= &wf_impl_operation_read
 };
 
+static char * wf_impl_filesystem_create_id(void)
+{
+    uuid_t uuid;
+    uuid_generate(uuid);
+    char id[UUID_STR_LEN];
+    uuid_unparse(uuid, id);
+
+	return strdup(id);
+}
+
+static bool wf_impl_filesystem_is_link_broken(char const * path, char const * id)
+{
+	bool result = false;
+
+	char buffer[PATH_MAX];
```

Comment:
Have a look at https://eklitzke.org/path-max-is-tricky.
I think this kind of path handling is not a good idea.

Reply:
I've remove all occurences of PATH_MAX. As replacement, a new function wf_create_string is used (asprintf is not used, since we should avoid using GNU extensions directly).